File: Python/Warmup-2/string_match.py
# Given 2 strings, a and b, return the number of the positions where they contain the same length 2 substring. 
# So "xxcaazz" and "xxbaaz" yields 3, since the "xx", "aa", and "az" substrings appear in the same place in both strings.

def string_match(a, b):
  count = 0
  short = min(len(a), len(b))
  for i in range(short - 1):
    if a[i:i+2] == b[i:i+2]:
      count += 1
  return count
    

File: Java/Warmup-2/stringMatch.java
/* Given 2 strings, a and b, return the number of the positions where they contain the same length 2 substring. 
So "xxcaazz" and "xxbaaz" yields 3, since the "xx", "aa", and "az" substrings appear in the same place in both strings. */

public int stringMatch(String a, String b) {
    int count = 0;
    int len = Math.min(a.length(), b.length()) - 1;
    for (int i = 0; i < len; i++) {
        if (a.substring(i, i + 2).equals(b.substring(i, i + 2)))
            count++;
    }
    return count;
}
